src/Analyzers/CSharp/Analyzers/RemoveRedundantElseStatement/RemoveRedundantElseStatementDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveRedundantElseStatement;
+
+/// <summary>
+/// Looks for code like:
+/// <code>
+///     if (a == b)
+///     {
+///         return c;
+///     }
+///     else
+///     {
+///         return d;
+///     }
+/// </code>
+/// 
+/// And offers to convert it to:
+///
+/// <code>
+///     if (a == b)
+///     {
+///         return c;
+///     }
+///     
+///     return d;
+/// </code>
+///
+/// For this conversion to make sense statement(s) in each <c>if</c> and <c>else if</c> must end with a jump
+/// (return, break, continue or throw) in order to preserve the program's correctness.
+/// </summary>
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal sealed class RemoveRedundantElseStatementDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
```

Comment:
📝 I worry about adding this as an analyzer and not a refactoring. There have been many cases where I included an explicit `else` for clarity, and I wouldn't want an analyzer undoing that work.

Reply:
I think both can work here. It's fine as an analyzer (it's off by default). 

When off, it would make sense to have as a refactoring too.  I can make such a change